libsolidity/ast/Types.cpp

```diff
@@ -2755,7 +2760,12 @@ bool FunctionType::operator==(Type const& _other) const
 
 BoolResult FunctionType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 {
-	return _convertTo.category() == category();
+	if (_convertTo.category() == category())
+	{
+		auto const& convertToType = dynamic_cast<FunctionType const&>(_convertTo);
```

Comment:
No change to `isImplicitlyConvertibleTo`?

Reply:
``isImplicitlyConvertibleTo`` uses ``equalExcludingStateMutability``, which requires equal ``kind``s, so I think it should be fine. At least as long we don't touch the case for referencing functions in base classes (as of this PR they are still ``Kind::Internal`` - both when implemented and when not).

Reply:
ok!